### ISSUE TITLE
Reading gain when auto and possible buffer errors

### ISSUE THREAD

**Issue by radiosd** [roger-/pyrtlsdr#20]
I'm using the rtlsdr library and your wrapper on a Raspberry Pi to investiagte DSP. It is a great help, and I think all is installed correctly, but I have seen some strange behaviour. 

1) if gain is auto, a call to read_gain generates an error that closes the device.

2) I find that using read_samples, some of the samples (the first ~1800) seem to be intially invalid and then left over from the previous read. Using rtl-sdr and piping to a file does not show this behaviour, so I do not think this is due to the osmocom library.

I'm wanting to be constructive, and I have some suggestions that I have tested and do seem to work. I am also completly new to GitHub. (signed up just to make this post) so I don't know how to pass on the ideas.

I could just clip out the changes and post them, there are only as few lines in reality, but there may be other more efficient ways.  Anvice on helping via github and the normal etiquet appreciated.

regards Richard

**Comment by nocarryr** [comment]
I have encountered this issue as well, but under different circumstances.

When gain is set to "0" the get_gain method receives the value and believes it's an error:
```python
def get_gain(self):
''' Get gain of tuner (in dB). '''

    result = librtlsdr.rtlsdr_get_tuner_gain(self.dev_p)
    if result == 0:
        self.close()
        raise IOError('Error when getting gain')

    return result/10
```

AFAIK, valid gain values should all be greater than or equal to 0.  Is that a correct assumption?

If so, I can send in a PR with the changes.

**Comment by roger-** [comment]
> AFAIK, valid gain values should all be greater than or equal to 0. Is that a correct assumption?

Doesn't appear to be. This was intentionally done as according to the API a [return value == 0 is an error](https://github.com/steve-m/librtlsdr/blob/master/include/rtl-sdr.h#L227). 

Have you come across a tuner where 0 dB is valid?

**Comment by nocarryr** [comment]
The function that you linked to is a setter though. The function directly below (rtlsdr_get_tuner_gain) shows "return 0 on success" in its comments.

Oddly, I do have "0" listed as a valid gain value in "valid_gains_db" on my RTL2832.

**Comment by roger-** [comment]
> The function that you linked to is a setter though

The documentation is above the prototype, so this the relevant part:

```c
/*!
 * Get actual gain the device is configured to.
 *
 * \param dev the device handle given by rtlsdr_open()
 * \return 0 on error, gain in tenths of a dB, 115 means 11.5 dB.
 */
RTLSDR_API int rtlsdr_get_tuner_gain(rtlsdr_dev_t *dev);
```

Maybe [raise the issue](https://github.com/steve-m/librtlsdr/issues) and see if this is a documentation/implementation bug?

**Comment by nocarryr** [comment]
Ah. Sorry I missed that.
Yes, it does seem to be better to raise it in the actual library repo.
Should I close the PR?

**Comment by roger-** [comment]
Sure, lets wait and see what they say first. Thanks!

**Comment by roger-** [comment]
Fixed (I think) in 8e71d1d9c765247b71fe96c697571fed1e04fb88.
